integration-tests/tracecontext/src/main/java/io/opentelemetry/Application.java

```diff
@@ -34,99 +41,93 @@
             ContextPropagators.create(W3CTraceContextPropagator.getInstance()));
   }
 
+  private enum ArmeriaGetter implements TextMapGetter<RequestHeaders> {
+    INSTANCE;
+
+    @Override
+    public Iterable<String> keys(RequestHeaders carrier) {
+      return carrier.names().stream().map(AsciiString::toString).collect(Collectors.toList());
+    }
+
+    @Nullable
+    @Override
+    public String get(@Nullable RequestHeaders carrier, String key) {
+      if (carrier == null) {
+        return null;
+      }
+      return carrier.get(key);
+    }
+  }
+
+  private enum ArmeriaSetter implements TextMapSetter<RequestHeadersBuilder> {
+    INSTANCE;
+
+    @Override
+    public void set(@Nullable RequestHeadersBuilder carrier, String key, String value) {
+      if (carrier == null) {
+        return;
+      }
+      carrier.add(key, value);
```

Comment:
I recall there is an issue filled to use set() instead of add() for propagation headers in order to avoid duplicates (eg when reusing a request instance). I know this is only test code, so not influencing a real instrumentation, but just to stay aligned with "best practice", why not?